plugins/database/mysql/mysql.go

```diff
@@ -316,6 +316,89 @@ func (m *MySQL) RotateRootCredentials(ctx context.Context, statements []string)
 	return m.RawConfig, nil
 }
 
+// SetCredentials uses provided information to set the password a user in the
+// database. Unlike CreateUser, this method requires a username be provided and
+// uses the name given, instead of generating a name. This is used for setting
+// the password of static accounts, as well as rolling back passwords in the
+// database in the event an updated database fails to save in Vault's storage.
+func (m *MySQL) SetCredentials(ctx context.Context, statements dbplugin.Statements, staticUser dbplugin.StaticUserConfig) (username, password string, err error) {
+	if len(statements.Rotation) == 0 {
+		return "", "", errors.New("empty rotation statements")
+	}
+
+	username = staticUser.Username
+	password = staticUser.Password
+	if username == "" || password == "" {
+		return "", "", errors.New("must provide both username and password")
+	}
+
+	// Grab the lock
+	m.Lock()
+	defer m.Unlock()
+
+	// Get the connection
+	db, err := m.getConnection(ctx)
+	if err != nil {
+		return "", "", err
+	}
+
+	// Start a transaction
+	tx, err := db.BeginTx(ctx, nil)
+	if err != nil {
+		return "", "", err
+	}
+	defer func() {
+		_ = tx.Rollback()
```

Comment:
I appreciate the explicitness here of using an underscore to show that the error is being ignored. I would suggest logging the error at the warn level, but the database plugins don't actually have access to a logger. Someday I would love to add one but it's low on the list.

Reply:
I agree, also `Rollback` is a bit confusing anyway, it seems to only be relevant if an error happened, and doesn't do anything if everything committed successfully? The defer of the rollback seems odd, I feel like a more explicit call to it in case of error would be better, but that's how the codebase is for `CreateUser` and I didn't want to get lost in the weeds tracking it down and figuring it out.

Reply:
FWIW the deferred rollback is a common pattern for transactions. Rollback is a no-op if the transaction is already committed